packages/web3.js-plugin-fee-estimation/src/plugin.ts

```diff
@@ -0,0 +1,337 @@
+      returnFormat ?? DEFAULT_RETURN_FORMAT
+    )
+  }
+
+  /**
+   * Retrieves the full semver version of GasPriceOracle
+   * @returns {Promise<string>} - The semver version
+   * @example
+   * const version = await web3.op.getVersion();
+   */
+  public async getVersion() {
+    return this._getPriceOracleContractInstance().methods.version().call()
+  }
+
+  /**
+   * Retrieves the amount of L2 gas estimated to execute {transaction}
+   * @param transaction - An unsigned web3.js {Transaction} object
+   * @param {{ blockNumber: BlockNumberOrTag, returnFormat: DataFormat }} [options={blockNumber: BlockTags.LATEST, returnFormat: DEFAULT_RETURN_FORMAT}] -
+   * An options object specifying what block to use for gas estimates and the web3.js format object that specifies how to format number and bytes values
+   * @returns {Promise<Numbers>} - The gas estimate as a BigInt by default, but {returnFormat} determines type
+   * @example
+   * const l2Fee: bigint = await getL2Fee(transaction);
+   * @example
+   * const numberFormat = { number: FMT_NUMBER.NUMBER, bytes: FMT_BYTES.HEX }
+   * const l2Fee: number = await getL2Fee(transaction, numberFormat);
+   */
+  public async getL2Fee<
+    ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT
+  >(
+    transaction: Transaction,
+    options?: {
+      blockNumber?: BlockNumberOrTag
+      returnFormat?: ReturnFormat
+    }
+  ) {
+    const [gasCost, gasPrice] = await Promise.all([
+      estimateGas(
+        this,
+        transaction,
+        options?.blockNumber ?? BlockTags.LATEST,
+        DEFAULT_RETURN_FORMAT
+      ),
+      this.getGasPrice(),
+    ])
+
+    return Web3.utils.format(
+      { format: 'uint' },
+      gasCost * gasPrice,
+      options?.returnFormat ?? DEFAULT_RETURN_FORMAT
+    )
+  }
+
+  /**
+   * Computes the total (L1 + L2) fee estimate to execute {transaction}
+   * @param transaction - An unsigned web3.js {Transaction} object
+   * @param {{ blockNumber: BlockNumberOrTag, returnFormat: DataFormat }} [options={blockNumber: BlockTags.LATEST, returnFormat: DEFAULT_RETURN_FORMAT}] -
+   * An options object specifying what block to use for gas estimates and the web3.js format object that specifies how to format number and bytes values
+   * @returns {Promise<Numbers>} - The estimated total fee as a BigInt by default, but {returnFormat} determines type
+   * @example
+   * const estimatedFees: bigint = await estimateFees(transaction);
+   * @example
+   * const numberFormat = { number: FMT_NUMBER.NUMBER, bytes: FMT_BYTES.HEX }
+   * const estimatedFees: number = await estimateFees(transaction, numberFormat);
+   */
+  public async estimateFees<
+    ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT
+  >(
+    transaction: Transaction,
+    options?: {
+      blockNumber?: BlockNumberOrTag
+      returnFormat?: ReturnFormat
+    }
+  ) {
+    const [l1Fee, l2Fee] = await Promise.all([
+      this.getL1Fee(transaction),
```

Comment:
Should this be passing in the options.blockNumber too?

Reply:
@roninjin10 Ideally yes, but specifying the `blockNumber` for contract calls is not supported by web3.js

Reply:
hmm I think that should likely bubble up to making this entire method not support blockNumber since it halfway supporting blockNumber makes this a bit of a leaky abstraction in that folks using it would need to be aware of that nuance

Reply:
Oh yea, that totally makes sense. Will remove it